### ISSUE TITLE
Make sure builds are passing

### ISSUE THREAD

**Issue by nozavroni** [deni-zen/csvelte#142]
Make sure all your build stuff is passing 
- [x] Travis - make sure build is passing for 5.5+
  - [x] PHP5.5+ 
  - [x] PHP7 - This build is failing due to test failure where "" != false (see build logs in comments for full report)
    - [x] Apparently [it isn't a good idea to name a class "Resource"](http://php.net/manual/en/migration70.incompatible.php#migration70.incompatible.other.classes), so I guess I'll need to change that (see #143).
- [x] Scrutinizer - get score above 8.5 
- [x] Coveralls - get coverage above 90%
- [x] RTD

**Comment by nozavroni** [comment]
```
Using worker: worker-linux-docker-95efbc6d.prod.travis-ci.org:travis-linux-4
system_info
Build system information
Build language: php
Build group: stable
Build dist: precise
Build id: 165477754
Job id: 165477756
travis-build version: c31595e05
Build image provisioning date and time
Thu Feb  5 15:09:33 UTC 2015
Operating System Details
Distributor ID: Ubuntu
Description:    Ubuntu 12.04.5 LTS
Release:    12.04
Codename:   precise
Linux Version
3.13.0-29-generic
Cookbooks Version
a68419e https://github.com/travis-ci/travis-cookbooks/tree/a68419e
GCC version
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
LLVM version
clang version 3.4 (tags/RELEASE_34/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
Pre-installed Ruby versions
ruby-1.9.3-p551
Pre-installed Node.js versions
v0.10.36
Pre-installed Go versions
1.4.1
Redis version
redis-server 2.8.19
riak version
2.0.2
MongoDB version
MongoDB 2.4.12
CouchDB version
couchdb 1.6.1
Neo4j version
1.9.4
RabbitMQ Version
3.4.3
ElasticSearch version
1.4.0
Installed Sphinx versions
2.0.10
2.1.9
2.2.6
Default Sphinx version
2.2.6
Installed Firefox version
firefox 31.0esr
PhantomJS version
1.9.8
ant -version
Apache Ant(TM) version 1.8.2 compiled on December 3 2011
mvn -version
Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T17:29:23+00:00)
Maven home: /usr/local/maven
Java version: 1.7.0_76, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "3.13.0-29-generic", arch: "amd64", family: "unix"
fix.CVE-2015-7547
$ export DEBIAN_FRONTEND=noninteractive
W: Size of file /var/lib/apt/lists/us.archive.ubuntu.com_ubuntu_dists_precise-backports_multiverse_source_Sources.gz is not what the server reported 5886 5888
W: Size of file /var/lib/apt/lists/dl.hhvm.com_ubuntu_dists_precise_main_binary-amd64_Packages.gz is not what the server reported 2047 2461
W: Size of file /var/lib/apt/lists/ppa.launchpad.net_travis-ci_zero-mq_ubuntu_dists_precise_main_binary-amd64_Packages.gz is not what the server reported 832 1195
W: Size of file /var/lib/apt/lists/ppa.launchpad.net_ubuntugis_ppa_ubuntu_dists_precise_main_binary-amd64_Packages.gz is not what the server reported 33653 36677
W: Size of file /var/lib/apt/lists/ppa.launchpad.net_ubuntugis_ppa_ubuntu_dists_precise_main_binary-i386_Packages.gz is not what the server reported 33699 36733
Reading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  libc-bin libc-dev-bin libc6-dev
Suggested packages:
  glibc-doc
The following packages will be upgraded:
  libc-bin libc-dev-bin libc6 libc6-dev
4 upgraded, 0 newly installed, 0 to remove and 248 not upgraded.
Need to get 8,840 kB of archives.
After this operation, 14.3 kB disk space will be freed.
Get:1 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc6-dev amd64 2.15-0ubuntu10.15 [2,943 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc-dev-bin amd64 2.15-0ubuntu10.15 [84.7 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc-bin amd64 2.15-0ubuntu10.15 [1,177 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu/ precise-updates/main libc6 amd64 2.15-0ubuntu10.15 [4,636 kB]
Fetched 8,840 kB in 0s (37.2 MB/s)
Preconfiguring packages ...
(Reading database ... 70845 files and directories currently installed.)
Preparing to replace libc6-dev 2.15-0ubuntu10.10 (using .../libc6-dev_2.15-0ubuntu10.15_amd64.deb) ...
Unpacking replacement libc6-dev ...
Preparing to replace libc-dev-bin 2.15-0ubuntu10.10 (using .../libc-dev-bin_2.15-0ubuntu10.15_amd64.deb) ...
Unpacking replacement libc-dev-bin ...
Preparing to replace libc-bin 2.15-0ubuntu10.10 (using .../libc-bin_2.15-0ubuntu10.15_amd64.deb) ...
Unpacking replacement libc-bin ...
Processing triggers for man-db ...
Setting up libc-bin (2.15-0ubuntu10.15) ...
(Reading database ... 70844 files and directories currently installed.)
Preparing to replace libc6 2.15-0ubuntu10.10 (using .../libc6_2.15-0ubuntu10.15_amd64.deb) ...
Unpacking replacement libc6 ...
Setting up libc6 (2.15-0ubuntu10.15) ...
Setting up libc-dev-bin (2.15-0ubuntu10.15) ...
Setting up libc6-dev (2.15-0ubuntu10.15) ...
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place
git.checkout
0.42s$ git clone --depth=50 --branch=releases/v0.2.1 https://github.com/nozavroni/csvelte.git nozavroni/csvelte
Cloning into 'nozavroni/csvelte'...
remote: Counting objects: 538, done.
remote: Compressing objects: 100% (274/274), done.
remote: Total 538 (delta 279), reused 462 (delta 251), pack-reused 0
Receiving objects: 100% (538/538), 217.13 KiB | 0 bytes/s, done.
Resolving deltas: 100% (279/279), done.
Checking connectivity... done.
$ cd nozavroni/csvelte
$ git checkout -qf 3d7a8514ca93568eedae6fbe93acb9b04d819599
This job is running on container-based infrastructure, which does not allow use of 'sudo', setuid and setguid executables.
If you require sudo, add 'sudo: required' to your .travis.yml
See https://docs.travis-ci.com/user/workers/container-based-infrastructure/ for details.
0.02s$ phpenv global 7.0 2>/dev/null
7.0 is not pre-installed; installing
7.93s
0.01s
0.02s$ phpenv global 7.0
1.31s$ composer self-update
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
You are already using composer version 1.2.1 (stable channel).
cache.1
Setting up build cache
$ export CASHER_DIR=$HOME/.casher
0.06s$ Installing caching utilities
0.00s
0.50sattempting to download cache archive
fetching releasesv0.2.1/cache-linux-precise-e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855--php-7.0.tgz
found cache
0.00s
1.27sadding /home/travis/.composer/cache to cache
$ php --version
PHP 7.0.11 (cli) (built: Sep 19 2016 20:53:22) ( ZTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.11, Copyright (c) 1999-2016, by Zend Technologies
    with Xdebug v2.4.1, Copyright (c) 2002-2016, by Derick Rethans
$ composer --version
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Composer version 1.2.1 2016-09-12 11:27:19
before_install
0.61s$ composer self-update
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
You are already using composer version 1.2.1 (stable channel).
install.1
0.00s$ export PATH="$HOME/.composer/vendor/bin:$PATH"
install.2
1.45s$ travis_retry ./travis/install.sh
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
  - Installing mikey179/vfsstream (v1.6.4)
    Loading from cache
  - Installing webmozart/assert (1.1.0)
    Loading from cache
  - Installing phpdocumentor/reflection-common (1.0)
    Loading from cache
  - Installing phpdocumentor/type-resolver (0.2)
    Loading from cache
  - Installing phpdocumentor/reflection-docblock (3.1.0)
    Loading from cache
  - Installing phpunit/php-token-stream (1.4.8)
    Loading from cache
  - Installing symfony/yaml (v3.1.4)
    Loading from cache
  - Installing sebastian/version (2.0.0)
    Loading from cache
  - Installing sebastian/resource-operations (1.0.0)
    Loading from cache
  - Installing sebastian/recursion-context (1.0.2)
    Loading from cache
  - Installing sebastian/object-enumerator (1.0.0)
    Loading from cache
  - Installing sebastian/global-state (1.1.1)
    Loading from cache
  - Installing sebastian/exporter (1.2.2)
    Loading from cache
  - Installing sebastian/environment (1.3.8)
    Loading from cache
  - Installing sebastian/diff (1.4.1)
    Loading from cache
  - Installing sebastian/comparator (1.2.0)
    Loading from cache
  - Installing phpunit/php-text-template (1.2.1)
    Loading from cache
  - Installing doctrine/instantiator (1.0.5)
    Loading from cache
  - Installing phpunit/phpunit-mock-objects (3.2.7)
    Loading from cache
  - Installing phpunit/php-timer (1.0.8)
    Loading from cache
  - Installing phpunit/php-file-iterator (1.4.1)
    Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.0)
    Loading from cache
  - Installing phpunit/php-code-coverage (4.0.1)
    Loading from cache
  - Installing phpspec/prophecy (v1.6.1)
    Loading from cache
  - Installing myclabs/deep-copy (1.5.4)
    Loading from cache
  - Installing phpunit/phpunit (5.5.4)
    Loading from cache
sebastian/global-state suggests installing ext-uopz (*)
phpunit/phpunit suggests installing phpunit/php-invoker (~1.1)
Generating autoload files
before_script.1
0.40s$ composer install
You are running composer with xdebug enabled. This has a major impact on runtime performance. See https://getcomposer.org/xdebug
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
before_script.2
1.71s$ ./travis/before_script.sh
--2016-10-06 09:59:07--  https://s3-us-west-2.amazonaws.com/csvelte/csvsampledata.tar.gz
Resolving s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)... 54.231.168.176
Connecting to s3-us-west-2.amazonaws.com (s3-us-west-2.amazonaws.com)|54.231.168.176|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2996629 (2.9M) [application/x-gzip]
Saving to: `csvsampledata.tar.gz'
100%[======================================>] 2,996,629   2.38M/s   in 1.2s    
2016-10-06 09:59:08 (2.38 MB/s) - `csvsampledata.tar.gz' saved [2996629/2996629]
files/
files/._.DS_Store
files/.DS_Store
files/._banklist-qsc-sm.csv
files/banklist-qsc-sm.csv
files/._banklist-qsc.csv
files/banklist-qsc.csv
files/._banklist.csv
files/banklist.csv
files/._datagovbldgrexus.csv
files/datagovbldgrexus.csv
files/._Demographic_Statistics_By_Zip_Code.csv
files/Demographic_Statistics_By_Zip_Code.csv
files/._iouzipcodes2011.csv
files/iouzipcodes2011.csv
files/._Most+Recent+Cohorts+(Scorecard+Elements).csv
files/Most+Recent+Cohorts+(Scorecard+Elements).csv
files/._sample.csv
files/sample.csv
files/._SampleCSVFile_1109kb.csv
files/SampleCSVFile_1109kb.csv
files/._SampleCSVFile_119kb.csv
files/SampleCSVFile_119kb.csv
files/._SampleCSVFile_11kb.csv
files/SampleCSVFile_11kb.csv
files/._SampleCSVFile_2kb.csv
files/SampleCSVFile_2kb.csv
files/._SampleCSVFile_556kb.csv
files/SampleCSVFile_556kb.csv
files/temp/
files/tempdeleteme.csv
files/._userssharedsdfratebrthsyaw1819raceethncty20002012.csv
files/userssharedsdfratebrthsyaw1819raceethncty20002012.csv
files/temp/deleteme.csv
2.80s$ ./travis/script.sh
PHPUnit 5.5.4 by Sebastian Bergmann and contributors.
Runtime:       PHP 7.0.11 with Xdebug 2.4.1
Configuration: /home/travis/build/nozavroni/csvelte/phpunit.xml.dist
.......................................................FF......  63 / 154 ( 40%)
............................................................... 126 / 154 ( 81%)
............................                                    154 / 154 (100%)
Time: 2.72 seconds, Memory: 10.00MB
There were 2 failures:
1) CSVelteTest\IO\StreamTest::testBufferStreamClose
Failed asserting that '' is false.
/home/travis/build/nozavroni/csvelte/tests/CSVelte/IO/StreamTest.php:682
2) CSVelteTest\IO\StreamTest::testBufferStreamDetach
Failed asserting that '' is false.
/home/travis/build/nozavroni/csvelte/tests/CSVelte/IO/StreamTest.php:690
FAILURES!
Tests: 154, Assertions: 421, Failures: 2.
The command "./travis/script.sh" exited with 1.
cache.2
store build cache
0.00s
0.27snothing changed, not updating cache
after_script
0.00s$ ./travis/after_script.sh
Done. Your build exited with 1.
```

**Comment by nozavroni** [comment]
I've decided to loosen the requirements for an initial release a bit starting now (so v0.2.1 will sort of count as like a new minor version for these purposes). I've outlined the process in the green binder, but to summarize, basically every initial minor release will be jam-packed with all kinds of new crap and I won't worry too much about things like code coverage, scrutinizer and other code quality metrics, documentation, etc. Then, in subsequent "patch" releases, I'll clean things up and slowly increase code coverage and quality metrics and docs until all that stuff is gravy. Then I'll start over with a new minor release. I will maintain this process until the interface stabilizes a little. Probably like v0.7 or 0.8. Somewhere around there... 

So... unless there is a build failure (Travis fails to build for PHP5.6 or PHP7), this release is a go...
